### ISSUE TITLE
Results not making it into data from useFind

### ISSUE THREAD

**Issue by ericuldall** [marshallswain/feathers-pinia#119]
Having this inconsistent behavior with useFind still. Here's the basic problem:

This works:
```
Event.useFind({ query: { $sort: { start: 1 } } });
```

This DOES NOT work:
```
Event.useFind({ query: { start: { $gte: new Date }, $sort: { start: 1 } } });
```

Ive confirmed the results are being correctly queried via the api, but for some reason they don't make it into the { data } property.

Any thoughts on this bug @marshallswain ?

**Comment by ericuldall** [comment]
So if I do this:
```
Event.useFind({ query: { start: { $gte: new Date }, $sort: { start: 1 } }, onServer: true, immediate: false });
```
Then it works... i'm not really sure what the different use cases are, maybe i've missed something in the docs.

**Comment by marshallswain** [comment]
>  i'm not really sure what the different use cases are, maybe i've missed something in the docs.

Yeah, the `onServer` attribute was poorly named.  That's a fail on my part.  I've renamed it to `paginateOnServer` in version 3.  Basically, when enabled, the `onServer` attribute disables live lists and only does reactive records.

The point of useFind is to use a single query to both pull in data from the api server and then a reactive list from the local store. It's tougher than you'd think, but in version 3 (which I'm working on right now) even the `paginateOnServer` mode will support live lists.  At least, as live as they can get.  Additional queries need to be made under the hood for some situations.

**Comment by marshallswain** [comment]
For more details about `onServer`'s behavior, every request made to the API server gets recorded in the store's `pagination` object, including the `id`s of each of the records.  When you put `useFind` into `onServer` mode, instead of making a `sift` filter query against the local store, it references the data for the pagination object and returns exactly the records that match that query.  This is why live lists don't currently work for `onServer` mode, because you have to make another query to the API server to know the state of the list after making a change.

In feathers-pinia v3, even in `paginateOnServer` mode we can avoid making additional queries if you already have the next page of data and you delete a record.  We can shift the record from the next page to the end of the current page and not have to make the query.  When we go to the next page, again, we'll need to fetch.

**Comment by ericuldall** [comment]
In feathers-vuex there was a feature that was a reactive list that was updated via the socket server. If a record was changed/added/removed via the api, the list in the client would reflect that in realtime. Is there some parody to this in feathers-pinia?

**Comment by marshallswain** [comment]
findInStore does that.  You can perform a separate query with find to pull in the initial data. The findInStore list will stay updated with real-time events.  useFind uses both under the hood.

**Comment by ericuldall** [comment]
Okay. Then I think there's definitely some disconnect in the data getting returned from useFind

**Comment by ericuldall** [comment]
@marshallswain Any idea why a useFind query would work when routing from one route to another, but does not work on the first page load in browser?

**Comment by marshallswain** [comment]
@ericuldall it sounds like a break in reactivity is occurring between the time that the page loads without data and the data comes in.

**Comment by marshallswain** [comment]
This is fixed in the latest 3.0 pre-release.
